apidoc/Titanium/UI/WebView.yml

```diff
@@ -244,20 +244,114 @@ methods:
   - name: setBasicAuthentication
     summary: |
         Sets the basic authentication for this web view to use on subsequent URl requests.
+    description: |
+        Persistence property used in this method is only supported on iOS platform and 
+        from SDK version 7.4.0.
     platforms: [android, iphone, ipad]
+
     parameters:
 
       - name: username
-        summary: Basic auth username.
+        summary: Basic auth user name.
         type: String
 
       - name: password
         summary: Basic auth password.
         type: String
 
+      - name: persistence
+        summary: Constants specify how long the credential will be kept.
+        type: Number
+        constants: Titanium.UI.CREDENTIAL_PERSISTENCE_*
+
   - name: stopLoading
     summary: Stops loading a currently loading page.
 
+  - name: startListeningToProperties
+    summary: Add native properties for observing for change. 
+    description: |
+        Some common properties are title, URL, estimatedProgress etc. 
+        See native KVO capability at https://developer.apple.com/documentation/webkit/wkwebview.
+        See example section "Listening to Web View properties in iOS" for usage.      
+    platforms: [iphone, ipad]
+    since: "7.4.0"
+    parameters:
+      - name: propertyList
+        summary: List of properties for listen.
+        type: Array
+
+  - name: stopListeningToProperties
+    summary: Remove native properties from observing.      
+    platforms: [iphone, ipad]
+    since: "7.4.0"
+    parameters:
+      - name: propertyList
+        summary: List of properties to remove.
+        type: Array
+
+  - name: takeSnapshot
+    summary: Takes a snapshot of the view's visible viewport.      
+    platforms: [iphone, ipad]
+    since: "7.4.0"
+    osver: {ios: {min: "11.0"}}
+    parameters:
+      - name: callback
+        summary: Function to call upon snapshot captured.
+        type: Callback<SnapshotResult>
```

Comment:
Does `takeSnapshot()` take a screenshot of the WebView's contents?

We already have a `toImage()` method for taking a screenshot, but it would include the WebView's border too which might not be desired. But we may want to double check if `toImage()` is still able to capture the WebView's contents or not. Would you mind double checking please?

And perhaps we should consider renaming `takeSnapshot()` to something else since `toImage()` already exists to avoid confusion between the two. Like `contentToImage()`? Or `pageToImage()`? Any other suggestions?

Reply:
It captures the actual visible area of the web-view (see [the Apple docs here](https://developer.apple.com/documentation/webkit/wkwebview/2873260-takesnapshotwithconfiguration)). @vijaysingh-axway Please check if we can have a < iOS 11 version as well and do the snapshot internally by overriding "toImage".

Reply:
takeSnapshot supported for iOS < 11 using 'toImage'.